File: abc359a/abc359a.hs
{-# LANGUAGE OverloadedStrings #-}
import Control.Monad(replicateM)
import Data.Text (Text)
import qualified Data.Text.IO as T

main :: IO ()
main = do
  n <- readLn
  solve <$> replicateM n T.getLine >>= print

solve :: [Text] -> Int
solve = length . filter (== "Takahashi")

File: abc359a/abc359a.ml
open Printf
open Scanf

let solve ls =
  List.filter (fun s -> s = "Takahashi") ls |> List.length

let () =
  let id x = x in
  scanf "%d " (fun n ->
    List.init n (fun _ -> scanf "%s " id)) |> solve |> printf "%d\n"
